Add AK-47 semi-auto rifle, Mini Draco pistol, AK magazines

carmakazi opened the pull request:
#### Summary

```SUMMARY: Content "Add AK-47 semi-auto rifle, Mini Draco pistol, AK magazines"```

#### Purpose of change

In anticipation of a coming realism pass on ammunition spawns, it is found that 7.62x39mm is currently a bit underrepresented in the game, as it is a fairly common sight in real life, though not as common as 5.56.  However, right now, the only civilian-legal, and therefore reasonably common, 7.62x39 rifle in the game is the SKS.  AK-variant rifles see a decent amount of popularity among American civilian shooters (though again, not as much as the AR-15), and so I believe these guns would be a good fit in the game.

#### Describe the solution

While there are many civilian AK pattern rifles available, the ones in this PR were chosen for the following reasons:
1) The WASR-10 has been available on the US market for some time, lending some to its availability.
2) The WASR-10 is reasonably priced, lending some to its availability.
3) The WASR-10 has at least a decent build quality, avoiding the quirks and issues of some other models currently available.
4) The WASR-10 has a distinct and concise name, in order to help players unfamiliar with guns differentiate it from an AK-47 assault rifle.
5) The Century Arms Draco pistol has the first two points going for it, but perhaps the build quality may be lacking a bit, and the lower durability and lack of the NEVER_JAMS flag reflects that.  Despite that, it is still a colloquially recognized and acquired weapon, even if for the wrong reasons sometimes.

As of this posting, the WASR-10 is a mid-low common drop, while the Draco is mid-low rare.  You may notice that the WASR is currently higher on the loot totem pole than the AR-15; I believe that's because the AR-15 is significantly underrepresented currently, but changing that falls outside the purview of this PR.

EDIT 4/01/2020: WASR-10 has been changed to generic AK-47 semi auto, and several new magazines have been added, including a 10, 20, and 75 round drum magazine.

#### Describe alternatives you've considered

Not adding it.

#### Testing

Test game loads with no issues, guns spawn into inventory and function as intended.

#### Additional

Please add 0.E content freeze and other appropriate tags.

anothersimulacrum commented:
Be sure to update generic guns for these weapons!

carmakazi commented:
> Be sure to update generic guns for these weapons!

I filed the WASR under the sporter rifle, and the Draco under the automagnum pistol, since there isn't a comparable pistol in the mod.  It's rare enough I don't think it should be a problem.

anothersimulacrum commented:
> since there isn't a comparable pistol in the mod
You can see what #37768 did, because that also added a rifle round pistol.
You could looking into adding some sort of rifle round pistol for these cases.
Or just leave what you have.

carmakazi commented:
That PR seems to also put the AR pistols under the sporter carbine.  It should work most of the time, but I wanted to avoid possible strange situations that would be caused by crossing gun types i.e. a rifle appearing where a pistol should be.  It's probably 12 one way, a dozen the other honestly, so I'm inclined to leave it be.

I wouldn't mind adding a GG analogue, but I think that'd be better left to the mod maintainer(s).  Not sure they'd even want it since these guns are such odd ducklings.

LaVeyanFiend commented on data/json/items/gun/762.json:
```diff
@@ -115,5 +115,80 @@
+    "reload_noise_volume": 10,
+    "name": { "str": "WASR-10" },
+    "description": "A semi-automatic-only civilian-legal Kalashnikov-pattern rifle, imported from Romania.  A valuable survivors' commodity.",
+    "weight": "3300 g",
```
```suggestion
    "weight": "3402 g",
```
https://www.centuryarms.com/wasr-144.html
7.5lb = 3402g, can round off if you want.

LaVeyanFiend commented on data/json/items/gun/762.json:
```diff
@@ -115,5 +115,80 @@
+    "name": { "str": "WASR-10" },
+    "description": "A semi-automatic-only civilian-legal Kalashnikov-pattern rifle, imported from Romania.  A valuable survivors' commodity.",
+    "weight": "3300 g",
+    "volume": "1750 ml",
```
```suggestion
    "volume": "4487 ml",
```
WASR-10 Image: https://www.centuryarms.com/wasr-144.html
Removal of mag for binarization: https://imgur.com/a/bKt44na
Surface area results(square in, conversion to square cm =631.986 cm^2): https://imgur.com/a/0mgxnGg
Width of 7.1cm using: https://www.gsg-waffen.com/site/index.php/en/ak47-tech-en

 (If @tenmillimaster has a better source on width this can be redone)

carmakazi replied:
Is this a new volume deriving method?  That seems awful high, it would put it almost up to the M107's volume.

I think it's in the width you got, they might have included the charging handle in that measurement.  Receiver thickness is ~35mm from what I'm reading, so with the furniture, safety lever, etc., maybe a bit more.  Unfortunately I do not own an AK-pattern rifle that I can put a caliper to.

LaVeyanFiend replied:
> Is this a new volume deriving method?

It's been used at least since August of last year(#33711), and has been used for all audits done by myself(https://github.com/CleverRaven/Cataclysm-DDA/pulls?q=is%3Apr+author%3ALaVeyanFiend+gun+volume+is%3Aclosed) or tenmillimaster.

>That seems awful high, it would put it almost up to the M107's volume.

The M107 is wrong, as are anything whose volume is 250ml * N.

>Receiver width of 35mm

👀 Source please, and I will redo the calculation for this & the Mini Draco.

carmakazi replied:
https://thefirearmsforum.s3.amazonaws.com/2011/09/127656_b8098900fc315bffa17c14246ff5412e.jpg

A work drawing for a homebuild receiver.  Bottom right-most dimension, 1.350" = 3.429cm.  Again, that would be slightly more when off-jutting bits like the safety are considered, even the rivets would add a bit, and I think the handguard is a little wider as well.

https://www.reddit.com/r/ak47/comments/3qzgpc/width_of_akms_underfolding_stock/

This guy claims that the major width of an underfolder stock is about 2.75" or 6.985cm, which significantly juts out from either side of the receiver, for a sanity check.

LaVeyanFiend replied:
I'll use the work drawing then.
```suggestion
    "volume": "2167 ml",
```

LaVeyanFiend commented on data/json/items/gun/762.json:
```diff
@@ -115,5 +115,80 @@
+    "reload_noise_volume": 10,
+    "name": { "str": "Mini Draco" },
+    "description": "Essentially a semi-auto AK with a short barrel and no stock, this unwieldy \"pistol\" has seen some popularity with civilian shooters and gangbangers alike.",
+    "weight": "2560 g",
```
```suggestion
    "weight": "2563 g",
```
You can ignore this if you want since it's just rounding, but the exact weight is 5.65lb or 2563g: https://www.centuryarms.com/mini-draco-pistol.html

LaVeyanFiend commented on data/json/items/gun/762.json:
```diff
@@ -115,5 +115,80 @@
+      [ "sights", 1 ],
+      [ "sling", 1 ],
+      [ "stock", 1 ],
+      [ "underbarrel", 1 ]
```
Most of these should be mounts, since WASR-10 doesn't come with Picatinny rails.

LaVeyanFiend commented on data/json/items/gun/762.json:
```diff
@@ -115,5 +115,80 @@
+      [ "sights", 1 ],
+      [ "sling", 1 ],
+      [ "stock", 1 ],
+      [ "underbarrel", 1 ]
```
ditto mounts

LaVeyanFiend commented on data/json/items/gun/762.json:
```diff
@@ -115,5 +115,80 @@
+    "name": { "str": "Mini Draco" },
+    "description": "Essentially a semi-auto AK with a short barrel and no stock, this unwieldy \"pistol\" has seen some popularity with civilian shooters and gangbangers alike.",
+    "weight": "2560 g",
+    "volume": "1250 ml",
```
```suggestion
    "volume": "2662 ml",
```
Mini Draco, overall length 17.50": https://www.centuryarms.com/mini-draco-pistol.html
Removing mag for binarization: https://imgur.com/a/as3lvUx
Surface area(square in =374.986 sq. cm): https://imgur.com/a/uZKEwEj
Using same width as WASR-10, unless you find a different source.

LaVeyanFiend replied:
Using the width you obtained from homemade receiver
```suggestion
    "volume": "1286 ml",
```

carmakazi commented:
I updated the stats based on your recommendations.  As for the mod slots, I changed the rail and underbarrel to require mounts (because no modular handguard), sights (because most optics would require an AK side rail adapter or other option, abstracted for game), and the Draco now requires a mount for a stock, as it has no provisions for a stock of any kind without an adapter or rear trunnion swap.  Since collapsing/folding stocks exist for AK type rifles and mods are generalized currently, I left the stock option open for the WASR.

tenmillimaster commented:
I believe referring to the semiauto AK as the WASR isn't the right choice. Nearly every firearms retailer markets their AK's under 'AK', or 'AK-47' names (with manufacturer or models appended), despite that being incorrect. The WASR should just be renamed 'AK', or 'AK-47'. The description can make it clear that it's a semiautomatic AKM clone of some flavor.

carmakazi commented:
> I believe referring to the semiauto AK as the WASR isn't the right choice. Nearly every firearms retailer markets their AK's under 'AK', or 'AK-47' names (with manufacturer or models appended), despite that being incorrect. The WASR should just be renamed 'AK', or 'AK-47'. The description can make it clear that it's a semiautomatic AKM clone of some flavor.

I'm not wholly on board, but I can understand wanting to generalize all the civilian AK models under one item like the AR-15 is and not have to split hairs over different models.  I'll make the change.

LaVeyanFiend commented on data/json/itemgroups/Weapons_Mods_Ammo/guns.json:
```diff
@@ -268,7 +270,8 @@
       { "item": "win70", "prob": 15, "charges-min": 0, "charges-max": 3 },
-      { "item": "1895sbl", "prob": 15, "charges-min": 0, "charges-max": 7 }
+      { "item": "1895sbl", "prob": 15, "charges-min": 0, "charges-max": 7 },
+      { "item": "wasr10", "prob": 35, "charges-min": 0, "charges-max": 30 }
```
```suggestion
      { "item": "aksemi", "prob": 35, "charges-min": 0, "charges-max": 30 }
```

LaVeyanFiend commented on data/json/itemgroups/Weapons_Mods_Ammo/guns.json:
```diff
@@ -291,7 +294,8 @@
       { "item": "win70", "prob": 15, "charges-min": 0, "charges-max": 0 },
-      { "item": "1895sbl", "prob": 15, "charges-min": 0, "charges-max": 0 }
+      { "item": "1895sbl", "prob": 15, "charges-min": 0, "charges-max": 0 },
+      { "item": "wasr10", "prob": 35, "charges-min": 0, "charges-max": 0 }
```
```suggestion
      { "item": "aksemi", "prob": 35, "charges-min": 0, "charges-max": 0 }
```

LaVeyanFiend commented on data/json/itemgroups/Weapons_Mods_Ammo/guns.json:
```diff
@@ -744,7 +748,8 @@
       { "item": "sten", "prob": 20, "charges-min": 0, "charges-max": 32 },
-      { "item": "sks", "prob": 10, "charges-min": 0, "charges-max": 10 }
+      { "item": "sks", "prob": 10, "charges-min": 0, "charges-max": 10 },
+      { "item": "wasr10", "prob": 5, "charges-min": 0, "charges-max": 30 }
```
```suggestion
      { "item": "aksemi", "prob": 5, "charges-min": 0, "charges-max": 30 }
```

LaVeyanFiend commented on data/mods/Generic_Guns/firearms/gg_firearms_migration.json:
```diff
@@ -255,7 +255,8 @@
       "sks",
-      "mosin44"
+      "mosin44",
+      "wasr10"
```
```suggestion
      "aksemi"
```

carmakazi commented:
In the spirit of #39156 I am looking at adding some AK magazines as well, including auditing the specs of the current ones.  Hold on merging.

tenmillimaster commented on data/json/items/gun/762.json:
```diff
@@ -115,5 +115,80 @@
+      [ "underbarrel mount", 1 ]
+    ],
+    "magazines": [ [ "762", [ "akmmag", "akmbigmag" ] ] ],
+    "flags": [ "NEVER_JAMS" ]
```
This (and the other AK) shouldn't have this flag.

tenmillimaster commented on data/json/items/obsolete.json:
```diff
@@ -1896,5 +1896,40 @@
     "difficulty": 8
+  },
+  {
+    "id": "akmbigmag",
```
Migrate these, don't stick them in obsolete. There's no need to have them hang around if you're going to provide replacements.

tenmillimaster commented on data/json/items/obsolete.json:
```diff
@@ -1896,5 +1896,40 @@
+    "flags": [ "MAG_BULKY" ]
+  },
+  {
+    "id": "akmmag",
```
Migrate these too.

tenmillimaster commented:
The draco will need a damage penalty. 
![image](https://user-images.githubusercontent.com/7764202/78253994-dbe24680-74ba-11ea-98f8-38b238a37230.png)

Night-Pryanik commented on data/json/items/gun/762.json:
```diff
@@ -115,5 +114,79 @@
+    "looks_like": "ar15",
+    "type": "GUN",
+    "reload_noise_volume": 10,
+    "name": { "str": "AK-47" },
```
Do we want to go with the "familiarity" way? I.e. most players recognize what AK-47 is.
Or with the "actuality" way? I.e. there is **no** IRL rifles named AK-47. The real rifles named simply "AK", without any numbers. Rifles named "AK-47" is a pilot batch. They were later officially taken into operation as "7,62mm Avtomat Kalashnikova (AK)".

tenmillimaster replied:
Generally, the AKM pattern rifles have 'AK' and '47' in their name. Given that the PC is (usually) an American, not a Cyrillic reading com-bloc aficionado, What retailers and enthusiasts refer to this pattern of rifle as is probably the right name. [For example](https://www.classicfirearms.com/long-guns/ak-47-for-sale/7.62x39/), [and here](https://www.atlanticfirearms.com/taxons/ak47-for-sale).

carmakazi replied:
My thoughts:

AK-47: colloquial and common marketing name but technically a misnomer.  Probably my choice of the 4, but I'm not a huge fan of any of them.
AK: too short of a name at 2 characters.
AK Sporter: a possible alternative, but to me it invokes images of thumbhole stocks and unconverted Saigas.
AK Semi-auto: a bit too janky in my opinion.

I'm going to wait until sometime tonight to let this discussion and other possible issues ferment before I make yet another commit.

carmakazi replied:
In the absence of any other input, I think I'm going to keep it as the AK-47.  Yes, I am aware of the history of the Kalashnikov and that no (actually used) rifle is really called the AK-47, but it is hard to neatly fit all of the different civilian models under one catch-all name like the AR-15 can.  We *could* make the WASR-10, the Zastava N-PAP, the Arsenal SGL-21, the Century C39V2, ad astra, to be technically correct, but that would just be useless bloat the same way adding Colt, BCM, Daniel Defense, Del-ton AR-15s would.  While realism and technical correctness seem to be appreciated where possible, this is one of those times where splitting hairs isn't going to get anywhere useful it seems.  The name might yank the chains of some "gun guys" playing the game, but we need to remember that we are in the minority, and that the majority of players I think will be fine with it so long as they read the description/stats and understand it isn't an automatic assault rifle.

tenmillimaster commented on data/json/items/magazine/762.json:
```diff
@@ -50,5 +66,39 @@
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "AK drum magazine" },
+    "description": "A bulky 75-round drum magazine made for 7.62x39  AK-pattern rifles.",
```
```suggestion
    "description": "A bulky 75-round drum magazine made for 7.62x39 AK-pattern rifles.",
```

ZhilkinSerg commented:
Please test your changes in-game and resolve following issues:

```
src/item_factory.cpp:1368 [const itype *Item_factory::find_template(const itype_id &) const] Missing item definition: akmbigmag

02:48:43.082 ERROR : src/item_factory.cpp:1368 [const itype *Item_factory::find_template(const itype_id &) const] Missing item definition: akmmag

02:48:43.083 ERROR : src/item_factory.cpp:1324 [void Item_factory::check_definitions() const] warnings for type cw-24hack:

magazine "akmbigmag" specified for "762" is not a magazine

magazine "akmmag" specified for "762" is not a magazine

02:48:43.083 ERROR : src/item_factory.cpp:1324 [void Item_factory::check_definitions() const] warnings for type cw-24:

magazine "akmbigmag" specified for "762" is not a magazine

magazine "akmmag" specified for "762" is not a magazine

02:48:43.083 ERROR : src/item_factory.cpp:1324 [void Item_factory::check_definitions() const] warnings for type svs-24c:

magazine "akmbigmag" specified for "762" is not a magazine

magazine "akmmag" specified for "762" is not a magazine

02:48:43.083 ERROR : src/item_factory.cpp:1324 [void Item_factory::check_definitions() const] warnings for type svs-24:

magazine "akmbigmag" specified for "762" is not a magazine

magazine "akmmag" specified for "762" is not a magazine

02:48:43.092 ERROR : src/item_factory.cpp:1324 [void Item_factory::check_definitions() const] warnings for type cw-24k:

magazine "akmbigmag" specified for "762" is not a magazine

magazine "akmmag" specified for "762" is not a magazine

02:48:43.128 ERROR : src/item_factory.cpp:1324 [void Item_factory::check_definitions() const] warnings for type cw-24m_hack:

magazine "akmbigmag" specified for "762" is not a magazine

magazine "akmmag" specified for "762" is not a magazine

02:48:43.128 ERROR : src/item_factory.cpp:1324 [void Item_factory::check_definitions() const] warnings for type cw-24m:

magazine "akmbigmag" specified for "762" is not a magazine

magazine "akmmag" specified for "762" is not a magazine
```

carmakazi commented:
> Please test your changes in-game and resolve following issues:

Twas an issue with Expanded Weapons still using the old AK mags.  Should be fine now, or the Travis build passed anyway.
